### PR TITLE
nginxModules.vod: 1.31 -> 1.32 and patch MAX_CLIPS

### PR DESCRIPTION
**Pull request by fleaz**
## Description of changes

[Changelog 1.31 -> 1.32](https://github.com/kaltura/nginx-vod-module/compare/1.31...1.32)

The change with MAX_CLIPS is necessary because the default limit is quite small and breaks some applications. For example Frigate also patches the file before compiling their own nginx: [build_nginx.sh#L31](https://github.com/blakeblackshear/frigate/blob/e0e8a6fcc915a5d12e76e33342fa634880cf0a16/docker/main/build_nginx.sh#L31)

And according to the upstream of nginx-vod, one should just patch this themself when a bigger value is needed.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Maybe put the reasoning into the commit message, so it isn't lost on GitHub at some point.

File: pkgs/servers/http/nginx/modules.nix [hunk 1]
```diff
@@ -985,8 +985,12 @@ let self = {
...
-      hash = "sha256-ZpeO8QWQ+fGkz08u/zFOq7vj4aHcodzSHNrc1SgGUyc=";
+      rev = "1.32";
+      hash = "sha256-ZpG0oj60D3o7/7uyE8AybCiOtncVe1Jnjaz22sIFypk=";
+      postFetch = ''
+        substituteInPlace $out/vod/media_set.h \
+          --replace "MAX_CLIPS (128)" "MAX_CLIPS (1024)"
+      '';
     };
 
     inputs = [ ffmpeg fdk_aac openssl libxml2 libiconv ];
```
Comment (mweinelt) [inline pkgs/servers/http/nginx/modules.nix:996]: I wonder whether we can get away with using `ffmpeg-headless`. That would save more than 200 MB.

```
$ nix path-info -Sh /nix/store/lymwbqc9hmfaknr1n0wlmcrdklfy9g54-ffmpeg-6.0-bin
/nix/store/lymwbqc9hmfaknr1n0wlmcrdklfy9g54-ffmpeg-6.0-bin	621.9M
$ nix path-info -Sh /nix/store/fxg8v7zfq6fan7fnmgsixlpizalqdrsw-ffmpeg-headless-6.0-bin
/nix/store/fxg8v7zfq6fan7fnmgsixlpizalqdrsw-ffmpeg-headless-6.0-bin	408.1M
```

Comment (fleaz) [reply pkgs/servers/http/nginx/modules.nix:996]: It at least it compiles without errors when using ffmpeg-headless. Currently deploying the new vod module to my server to check if it works 🤞🏻

Comment (fleaz) [reply pkgs/servers/http/nginx/modules.nix:996]: Looks good. At least the functionality that Frigate uses from the VOD module, is not affected by switching to ffmpeg-headless :)

**Pull request merged**